autogluon/utils/tabular/ml/models/lgb/lgb_model.py

```diff
@@ -74,6 +74,15 @@ def _fit(self, X_train=None, y_train=None, X_val=None, y_val=None, dataset_train
         gc.collect()
 
         num_boost_round = params.pop('num_boost_round', 1000)
+        dart_retrain = params.pop('dart_retrain', False)  # Whether to retrain the model to get optimal iteration if model is trained in 'dart' mode.
```

Comment:
For both of these hyperparams, I don't think we should have specify the default value here?  Shouldn't we just ensure all default values come from `hyperparameters/parameters.py`, so that they're very clearly documented?

Reply:
dart_retrain is very niche, and probably will only be for internal usage. I agree in general however, and I think we can maybe clean that up. Do you suggest that if num_boost_round doesn't exist at this point in the code to throw an exception?

Reply:
Yep could just do `params['num_boost_round']` instead of `.pop()`